src/color_laws.jl

```diff
@@ -127,13 +127,87 @@ function cal00_invum(x::Real, Rv::Real)
     if x > 1 / 0.12
         return 0.0x
     elseif x > 1 / 0.63
-        k = 2.659 * (((0.011x - 0.198)x + 1.509)x - 2.156)
+        k = 2.659 * Poly([-2.156, 1.509, -0.198, 0.011])(x)
     elseif x >  1 / 2.2
-        k = 2.659 * (1.040x - 1.857)
+        k = 2.659 * Poly([-1.857, 1.040])(x)
     else
         return 0.0x
     end
 
     return 1.0 + k / Rv
 
 end
+
+
+"""
+    gcc09(λ::Real, Rv=3.1)
+    gcc09(λ::Quantity, Rv=3.1)
+
+Gordon, Cartledge, & Clayton (2009) dust law.
+
+This model applies to the UV spectral region all the way to 909.09 Å.
+This model was not derived for the optical or NIR.
+
+If `λ` is a `Unitful.Quantity` it will be automatically converted to Å and the
+returned value will be `UnitfulAstro.mag`.
+"""
+function gcc09(λ::Real, Rv = 3.1)
+    x = aa_to_invum(λ)
+    return gcc09_invum(x, Rv)
+end
+
+gcc09(λ::Quantity, Rv::Real = 3.1) = gcc09(ustrip(u"angstrom", λ), Rv) * u"mag"
+
+function gcc09_invum(x::Real, Rv::Real)
+
+    if 3.3 <= x <= 11.0  # NUV
+        a = 1.894 - 0.373 * x - 0.0101 / ((x - 4.57)^2 + 0.0384)
+        b = -3.490 + 2.057 * x + 0.706 / ((x - 4.59)^2 + 0.169)
+    else
+        return 0.0x
+    end
+    if 5.9 <= x <= 11.0  # far-NUV
+        y = x - 5.9
+        a += Poly([0.0, 0.0, -0.110, -0.0100])(y)
+        b += Poly([0.0, 0.0, 0.531, 0.0544])(y)
+    end
```

Comment:
again, see my comments above on using `@evalpoly`

Reply:
updated them to `@evalpoly`!